src/types/message-types.ts

```diff
@@ -70,3 +70,17 @@ export type RangeFilter = (GT | LT) & Partial<GT> & Partial<LT>;
 export type Filter = {
   [property: string]: EqualFilter | OneOfFilter | RangeFilter
 };
+
+export type Pagination = {
+  messageCid?: string
```

Comment:
Should we rename this to something like `cursor`? Even though a good choice is to use `messageCid`, it remains an implementation detail and another implementation may choose to use something else.

Reply:
I originally had it as cursor but wanted it to be more descriptive.

That's also an interesting viewpoint, I didn't really consider this as an implementation detail, would we want different implementations using a different type of cursor value?

Initially I didn't think so, it felt like the same results should be expected per the message regardless of implementation.

I'm still kind of leaning in that direction personally. This is why `messageCid` made sense, as every message in the system has a unique one.

But definitely open to changing my mind.

Reply:
Given this a bit more thought, and I still feel like while I agree the returned results should be the same, the marker/cursor/watermark used to denote progress should be left to the implementor, `messageCid` is really a byproduct/necessity of LevelDB implementation.